### PR TITLE
Extract button base

### PR DESCRIPTION
**Pull request by manuelpuyol**
Initial discussion: https://github.com/primer/view_components/pull/445#issuecomment-818828571

# Summary

We always want to use a component when rendering a Button since we can share some behaviors (like not adding `type='button'` when using a `summary` tag). Our current button's problem is that it always adds the `btn` style, which is needed in most cases, but not all.

We have some buttons like `CloseButton`, `ButtonMarketing` and `HiddenTextExpander` that don't need all of our `ButtonComponent` styles, but they do need the shared behaviors.

# Solution

I decided to extract a `BaseButton`, which is basically an unstyled button. It holds the logic for possible `types` and `tags` and turning buttons into `blocks`. In the future, we could add some accessibility features too!
The new component should be used any time you need a button without the need for `btn`, `schemes`, and `variants`.

### CODE REVIEW

File: app/components/primer/base_button.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use BaseButton to render an unstyles `<button>` tag that can be customized.
```
Comment (joelhawksley) [inline app/components/primer/base_button.rb:4]:
```suggestion
  # Use BaseButton to render an unstyled `<button>` tag that can be customized.
```

File: app/components/primer/base_button.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
```
Comment (joelhawksley) [inline app/components/primer/base_button.rb:1]: Do you think we should be exposing this component as a public API? IMO we should keep it private, at least to start.

Comment (manuelpuyol) [reply app/components/primer/base_button.rb:1]: I'm fine keeping it private, that's why I didn't add it to the docs sidebar 😄

Comment (joelhawksley) [reply app/components/primer/base_button.rb:1]: Ah, cool. 👍🏻

**Review by joelhawksley — APPROVED** [review]

**Pull request merged**